nextest-runner/src/reporter/structured/libtest.rs

```diff
@@ -0,0 +1,529 @@
+// Copyright (c) The nextest Contributors
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! libtest compatible output support
+//!
+//! Before 1.70.0 it was possible to send `--format json` to test executables and
+//! they would print out a JSON line to stdout for various events. This format
+//! was however not intended to be stabilized, so 1.70.0 made it nightly only as
+//! intended. However, machine readable output is immensely useful to other
+//! tooling that can much more easily consume it than parsing the output meant
+//! for humans.
+//!
+//! Since there already existed tooling using the libtest output format, this
+//! event aggregator replicates that format so that projects can seamlessly
+//! integrate cargo-nextest into their project, as well as get the benefit of
+//! running their tests on stable instead of being forced to nightly.
+//!
+//! This implementation will attempt to follow the libtest format as it changes,
+//! but the rate of changes is quite low (see <https://github.com/rust-lang/rust/blob/master/library/test/src/formatters/json.rs>)
+//! so this should not be a big issue to users.
```

Comment:
I'd add a clarifying comment that in that case, the version number will be bumped up.